app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,35 @@
+class MarketsController < ApplicationController
+  before_action :set_nav
+  before_action :set_market, only: [:show, :edit, :update]
+
+  def index
+    @markets = Market.all
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    market = Market.create(market_params)
+    redirect_to market
+  end
```

Comment:
What happens if something goes wrong when you create a new `Market`? It'd be good to see you handling that possibility explicitly rather than letting it fail quietly.